python/GafferArnoldTest/IECoreArnoldPreviewTest/RendererTest.py

```diff
@@ -226,6 +226,69 @@ def testShaderNames( self ) :
 			self.assertEqual( len( shaders ), 4 )
 			self.assertEqual( len( set( [ arnold.AiNodeGetName( s ) for s in shaders ] ) ), 4 )
 
+	def testShaderNodeConnectionType( self ) :
+
+		r = GafferScene.Private.IECoreScenePreview.Renderer.create(
+			"IECoreArnold::Renderer",
+			GafferScene.Private.IECoreScenePreview.Renderer.RenderType.SceneDescription,
+			self.temporaryDirectory() + "/test.ass"
+		)
+
+		scalarColorShader = IECore.ObjectVector( [
+			IECore.Shader( "image", parameters = { "__handle" : "scalarColorSource" } ),
+			IECore.Shader( "lambert", parameters = { "Kd_color" : "link:scalarColorSource", "__handle" : "scalarColorTarget" } ),
+		] )
+
+		scalarNodeShader = IECore.ObjectVector( [
+			IECore.Shader( "gobo", parameters = { "__handle" : "scalarNodeSource" } ),
+			IECore.Shader( "persp_camera", parameters = { "filtermap" : "link:scalarNodeSource", "__handle" : "scalarNodeTarget" } ),
```

Comment:
This isn't a shader node, so it seems a bit odd odd to use it here. I guess there are no other alternatives which have such a link? Perhaps we could add a comment just explaining why we're abusing the node type slightly.